### PR TITLE
Mouse View feature. Issue 28889

### LINKED ISSUE #28889: Please restore Mouse View functionality to the sidebar.
Please restore Mouse View functionality to the sidebar.

Prior to the new customisable sidebar layout Mouse View used to appear whenever the mouse was moved from the sidebar into the main window, temporarily covering the compass and part of the message log.
It automatically disappeared upon movement of the player character, or the mouse returning to the sidebar.

If the old method of displaying it isn't possible under the new layout please could it be added as an optional panel in the sidebar.
(It could default to off, if you are concerned about preventing clutter in the default view)

I'm aware we can still use mouse look via opening the Look Around pop-up menu with ; or x, however this adds unnecessary repeated keystrokes, including having to press ESC to close the pop-up, each time we start and finish looking around.

**Versions and configuration(please complete the following information):**
 - OS: Windows 7
 - Game Version: 0.D-841-g462722b
 - Graphics version: Tiles

### PR DESCRIPTION
**Pull request by RoyBerube**
#### Summary
SUMMARY: Interface "Mouse view feature working again".
Fixes #28889

#### Purpose of change
Feature that was working in 0.D stable has been missing.

#### Describe the solution
Leveraged existing code to display location contents where the mouse is hovering. Mouse view is activated on mouse movement and deactivated on any key input.

#### Testing
Tested on Windows only. Tried with panel window starting in various vertical locations with no problems.

#### Additional context
Behaves much like the old mouse view feature. Currently the start location is fixed to the top of the info panel area. User cannot adjust the location of the window.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -2165,16 +2165,23 @@ int game::inventory_item_menu( int pos, int iStartX, int iWidth,
 bool game::handle_mouseview( input_context &ctxt, std::string &action )
 {
     cata::optional<tripoint> liveview_pos;
+    auto& mgr = panel_manager::get_manager();
```
Comment (ZhilkinSerg) [inline src/game.cpp]:
```suggestion
    panel_manager &mgr = panel_manager::get_manager();
```

File: src/game.cpp [hunk 2]
```diff
@@ -2165,16 +2165,23 @@ int game::inventory_item_menu( int pos, int iStartX, int iWidth,
...
+    const bool sidebar_right = get_option<std::string>("SIDEBAR_POSITION") == "right";
+    int width = sidebar_right ? mgr.get_width_right() : mgr.get_width_left();
     do {
         action = ctxt.handle_input();
         if( action == "MOUSE_MOVE" ) {
             const cata::optional<tripoint> mouse_pos = ctxt.get_coordinates( w_terrain );
             if( mouse_pos && ( !liveview_pos || *mouse_pos != *liveview_pos ) ) {
                 liveview_pos = mouse_pos;
                 liveview.show( *liveview_pos );
+                draw_panels(true);
```
Comment (ZhilkinSerg) [inline src/game.cpp]:
```suggestion
                draw_panels( true );
```

File: src/game.cpp [hunk 3]
```diff
@@ -2165,16 +2165,23 @@ int game::inventory_item_menu( int pos, int iStartX, int iWidth,
...
+                catacurses::window& w = catacurses::newwin(TERMY/2, width, point(sidebar_right ? TERMX - width : 0, 0));
```
Comment (ZhilkinSerg) [inline src/game.cpp]:
```suggestion
                catacurses::window& w = catacurses::newwin( TERMY/2, width, point( sidebar_right ? TERMX - width : 0, 0 ) );                
```

File: src/game.cpp [hunk 4]
```diff
@@ -2165,16 +2165,23 @@ int game::inventory_item_menu( int pos, int iStartX, int iWidth,
...
     do {
         action = ctxt.handle_input();
         if( action == "MOUSE_MOVE" ) {
             const cata::optional<tripoint> mouse_pos = ctxt.get_coordinates( w_terrain );
             if( mouse_pos && ( !liveview_pos || *mouse_pos != *liveview_pos ) ) {
                 liveview_pos = mouse_pos;
                 liveview.show( *liveview_pos );
+                draw_panels(true);
+                catacurses::window& w = catacurses::newwin(TERMY/2, width, point(sidebar_right ? TERMX - width : 0, 0));                
+                g->liveview.draw(w, TERMY/2);
```
Comment (ZhilkinSerg) [inline src/game.cpp]:
```suggestion
                g->liveview.draw( w, TERMY/2 );
```

File: src/game.cpp [hunk 5]
```diff
@@ -2165,16 +2165,23 @@ int game::inventory_item_menu( int pos, int iStartX, int iWidth,
...
             } else if( !mouse_pos ) {
                 liveview_pos.reset();
                 liveview.hide();
+                draw_panels(true);
```
Comment (ZhilkinSerg) [inline src/game.cpp]:
```suggestion
                draw_panels( true );
```

File: src/live_view.cpp [hunk 6]
```diff
@@ -51,10 +51,12 @@ int live_view::draw( const catacurses::window &win, const int max_height )
     // be a different code path here that works for ncurses.
     const int original_height = win.get<cata_cursesport::WINDOW>()->height;
     win.get<cata_cursesport::WINDOW>()->height = live_view_box_height;
+    g->draw_panels();   
 #endif
 
     draw_border( win );
     center_print( win, 0, c_white, _( "< <color_green>Mouse View</color> >" ) );
+    wrefresh(win);
```
Comment (ZhilkinSerg) [inline src/live_view.cpp]:
```suggestion
    wrefresh( win );
```

**Comment by ZhilkinSerg** [comment]
Changed files need to be astyled (I've made proposals, but could miss something, so you better run astyle by yourself).

Summary line format is `SUMMARY: Interface "Mouse view feature working again"`.

If you want to close issue automatically, use `Fixes #28889`.

**Comment by RoyBerube** [comment]
Ok I've formatted with Astyle and managed to screw up by adding 40+ unwanted commits. Clearly I don't understand Github yet. Going to sleep and feel shame for now.

**Comment by RoyBerube** [comment]
Closing and will be resubmitting.

**Pull request closed without merging**